GitUI/CommandsDialogs/FormFileHistory.cs

```diff
@@ -271,8 +276,33 @@ private void LoadFileHistory()
 
                     var lines = Module.GitExecutable.GetOutputLines(args, outputEncoding: GitModule.LosslessEncoding);
 
+                    ObjectId currentObjectId = null;
                     foreach (var line in lines.Select(GitModule.ReEncodeFileNameFromLossless))
                     {
+                        if (line.Length == 0)
+                        {
+                            // end of filenames
+                            currentObjectId = null;
+                            continue;
+                        }
+
+                        if (line.StartsWith(startOfObjectId))
+                        {
+                            currentObjectId = null;
+                            if (line.Length < ObjectId.Sha1CharCount + startOfObjectId.Length
+                                || !ObjectId.TryParse(line.Substring(startOfObjectId.Length, ObjectId.Sha1CharCount), 0, out currentObjectId))
```

Comment:
We should add a ReadOnlySpan<char> Parse method...

Reply:
There is a readOnlySpan in #9143 added after you approved.
(That PR got too big due to changes, now this PR changes something fundamental anyway...)

I will rebase #9143 on this PR (got the tests working too)